app/src/main/java/app/revanced/manager/ui/component/SegmentedButton.kt

```diff
@@ -28,41 +30,54 @@ fun RowScope.SegmentedButton(
     icon: Any,
     iconDescription: String? = null,
     text: String,
-    onClick: () -> Unit
+    onClick: () -> Unit,
+    enabled: Boolean = true
 ) {
-    Column(
-        horizontalAlignment = Alignment.CenterHorizontally,
-        verticalArrangement = Arrangement.spacedBy(8.dp, Alignment.CenterVertically),
-        modifier = Modifier
-            .clickable(onClick = onClick)
-            .background(MaterialTheme.colorScheme.surfaceColorAtElevation(2.dp))
-            .weight(1f)
-            .padding(vertical = 20.dp)
-    ) {
-        when (icon) {
-            is ImageVector -> {
-                Icon(
-                    imageVector = icon,
-                    contentDescription = iconDescription,
-                    tint = MaterialTheme.colorScheme.primary
-                )
-            }
+    val contentColor = if (enabled)
+        MaterialTheme.colorScheme.primary
+    else
+        MaterialTheme.colorScheme.onSurface.copy(0.38f)
 
-            is Painter -> {
-                Icon(
-                    painter = icon,
-                    contentDescription = iconDescription,
-                    tint = MaterialTheme.colorScheme.primary
+    CompositionLocalProvider(LocalContentColor provides contentColor) {
+        Column(
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.spacedBy(8.dp, Alignment.CenterVertically),
+            modifier = Modifier
+                .clickable(enabled = enabled, onClick = onClick)
+                .background(
+                    if (enabled)
+                        MaterialTheme.colorScheme.surfaceColorAtElevation(2.dp)
+                    else
+                        MaterialTheme.colorScheme.onSurface.copy(0.12f)
                 )
+                .weight(1f)
+                .padding(vertical = 20.dp)
+        ) {
+            when (icon) {
+                is ImageVector -> {
+                    Icon(
+                        imageVector = icon,
+                        contentDescription = iconDescription,
+                        //tint = MaterialTheme.colorScheme.primary,
```

Comment:
Why was this changed? Why is the comment left here?

Reply:
The content color should be gray when it is disabled. Im providing the color with CompositionLocalProvider now

Reply:
Fair enough. The comment should be removed though